From cfb4240a1b690867f9c0495ce0c267a77d7195a3 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Fri, 12 May 2023 15:21:36 +0530
Subject: [PATCH 3/3] ath12k: fix skb_cb corruption in tx data path

skb_cb memory shared across mac80211 and ath12k in tx path. In MLO multicast tx path,
mac80211 skb private data struture get written after populating the link_id information
in ath12k skb private data structure.

Fix it by avoid the mac80211 skb private access.

Fixes: I0ef93dc0ca8458 (ath12k: fix encrypted MLO Tx multicast issue)

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index bb0bccb..ca8f7fe 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -8761,7 +8761,6 @@ static void ath12k_mac_op_tx(struct ieee80211_hw *hw,
 			/* upper layer would not have handled the keys so
 			 * handle it here */
 			skb_cb = ATH12K_SKB_CB(msdu_copied);
-			info = IEEE80211_SKB_CB(msdu_copied);
 			skb_cb->link_id = link_id;
 
 			spin_lock_bh(&tmp_ar->ab->base_lock);
@@ -8776,7 +8775,6 @@ static void ath12k_mac_op_tx(struct ieee80211_hw *hw,
 			if (key) {
 				skb_cb->cipher = key->cipher;
 				skb_cb->flags |= ATH12K_SKB_CIPHER_SET;
-				info->control.hw_key = key;
 
 				hdr = (struct ieee80211_hdr *)msdu_copied->data;
 				if (!ieee80211_has_protected(hdr->frame_control))
-- 
2.17.1

